docs/the-new-architecture/pillars-cxxturbomodules.md

```diff
@@ -0,0 +1,698 @@
+
+And then re-using it in `OnLoad.cpp` for Android and `AppDelegate.mm` for iOS, e.g., via:
+
+```cpp
+static facebook::react::AppTurboModuleProvider appTurboModuleProvider;
+return appTurboModuleProvider.getTurboModule(name, jsInvoker);
+```
+
+in the corresponding functions.
+
+## Supporting custom types
+
+By default C++ Turbo Native Modules support [bridging functionality](https://github.com/facebook/react-native/tree/main/ReactCommon/react/bridging) for most `std::` standard types.
+
+If you want to add support for new / custom types in your app / library, you only need to provide the necessary `bridging` header file.
+
+### Example: Int64
+
+C++ Turbo Native Modules don't support `int64_t` numbers yet - because JavaScript doesn't support numbers greater `2^53`. We can't represent numbers > `2^53` as JavaScript `number`'s - but we can represent them as JavaScript `string`'s and automatically convert (aka. `bridge`) them to C++ `int64_t`'s by creating a custom Bridging header file:
+
+```cpp Int64.h
+#pragma once
+
+#include <react/bridging/Bridging.h>
+
+namespace facebook::react {
+
+template <>
+struct Bridging<int64_t> {
+  static int64_t fromJs(jsi::Runtime &rt, const jsi::String &value) {
+    try {
+      size_t pos;
+      auto str = value.utf8(rt);
+      auto num = std::stoll(str, &pos);
+      if (pos != str.size()) {
+        throw std::invalid_argument("Invalid number"); // don't support alphanumeric strings
+      }
+      return num;
+    } catch (const std::logic_error &e) {
+      throw jsi::JSError(rt, e.what());
+    }
+  }
+
+  static jsi::String toJs(jsi::Runtime &rt, int64_t value) {
+    return bridging::toJs(rt, std::to_string(value));
+  }
+};
+
+} // namespace facebook::react
+```
+
+Now you can add the following function to your JavaScript spec:
+
+<Tabs groupId="turbomodule-specs" defaultValue={constants.defaultJavaScriptSpecLanguages} values={constants.javaScriptSpecLanguages}>
+<TabItem value="typescript">
+
+```typescript title="NativeSampleModule.ts"
+   // ...
+   readonly passLargeNumber: (input: string) => number;
+   // ..
+```
+
+</TabItem>
+<TabItem value="flow">
+
+```js title="NativeSampleModule.js"
+   // ...
+   +passLargeNumber: (input: string) => number;
+   // ..
+```
+
+</TabItem>
+</Tabs>
+
+Declare it in your `NativeSampleModule.h` file and include the `Int64.h` header file:
+
+```cpp
+//...
+#include "Int64.h"
+//...
+int32_t passLargeNumber(jsi::Runtime& rt, int64_t input);
+```
+
+And implement it in `NativeSampleModule.cpp`:
+
+```cpp
+//...
+#include <cmath>
+//...
+int32_t NativeSampleModule::passLargeNumber(jsi::Runtime& rt, int64_t input) {
+    return std::cbrt(input);
+}
+```
+
+In your app you can call this new native function via:
+
+```js
+<Section title="Cxx TurboModule">
+  NativeSampleModule.passLargeNumber(...) ={' '}
+  {JSON.stringify(
+    NativeSampleModule.passLargeNumber('9223372036854775807')
+  )}
+</Section>
+```
+
+which should return `2097152`.
+
+### Any custom type
```

Comment:
The custom type section is continuing here... So I would probably create a different page for it. In this way, we may have a self-contained C++ TM guide and a guide on how extend the JSI typing system and the two would be easier to search for.

Reply:
This pieace is also extremely important in general for the New Architecture. It can be used as a workaround for some types that are not supported by the Codegen yet, so it definitely deserves its own page, imho.

Reply:
Where would you place this 'new' page and how should we call it?
<img width="341" alt="image" src="https://user-images.githubusercontent.com/484044/205750429-89177356-6a6a-4d84-bc76-d1246d7ee786.png">

Reply:
What about: `Supporting Custom C++ Types`?
I'm thinking whether we should have a section between `Pillars` and `Backward Compatibility` called `C++` where we can put all the C++ only stuff. What do you think?